Snakefile

```diff
@@ -98,6 +98,23 @@ if "inputs" not in config:
     ))
     sys.exit(1)
 
+# Check for configs using `*_exposure` which is no longer handled by our analysis pipeline.
+if "exposure" in config:
+    logger.warning("Your config specifies entires for 'exposure'. This is no longer used by this pipeline and can be removed.")
+if "skip_travel_history_adjustment" in config:
+    if config["skip_travel_history_adjustment"] is True:
+        logger.warning("Your config specifies 'skip_travel_history_adjustment=True'. This is now always the case, and thus this parameter can be removed.")
+    else:
+        logger.error("Your config specifies 'skip_travel_history_adjustment=False' however travel history (exposure) has been removed from this pipeline!")
+        sys.exit(1)
+for name,trait_info in config.get("traits", {}).items():
+    fatal = False
+    if len([col for col in trait_info.get("columns", []) if "_exposure" in col])>0:
+        logger.error(f"Build {name} is asking to reconstruct ancestral traits for exposure metadata, however the pipeline no longer uses exposure values.")
+        fatal = True
+    if fatal:
+        sys.exit(1)
```

Comment:
This very considerate! Do you think we could make this a major (breaking) release, document these changes well, and broadcast the changes on social media instead of keeping these checks in here?

Or, can we at least plan to remove these checks at the next major release after this one?

Reply:
✔️ We can make this a major release. What is the general objection to keeping these checks around? (We can move them to a special `.smk` if we want to keep the main snakefile clean.) My gut tells me that a sizeable fraction of users won't read changelogs, no matter how well we write them!

Reply:
I agree with James here, I think most people don't read changelogs :) We could keep these checks in for a release or a few, then let them go :)

Reply:
It's definitely a balance between user-friendliness and code bloat. My main arguments for avoiding this kind of code are 1) major releases are designed to allow us to make backward-incompatible (breaking) changes to the workflow to avoid maintaining legacy code or notifications about legacy code and 2) if we commit to supporting warnings for deprecated features (under the assumption that some users never updated their workflow configs), we will never be able to remove this code even when it isn't relevant a year from now.

On the one hand, the kindest move would be to include the warnings in this major release and then remove them in the next major release. On the other hand, this is a task we'd need to keep on our stack and remember for the next release. I'd bet that most people don't regularly update their workflow much less read the change logs. But those people that do update regularly, probably will check the change log.

If people don't read the change log or update their workflow, how can we improve awareness of these changes? I don't think we tweet from @nextstrain when we release new software (this workflow, auspice, augur, nextclade), but maybe we should. I would love for others to be as excited to read our change logs to find the cool new features we offer as I am to read Snakemake's and other tools' change logs. 🤓